### PR TITLE
implemented on touch listener.

### PR DESCRIPTION
**Pull request by mrkamiali**
Hi,@PierfrancescoSoffritti I have implemented ontouch listener on the player,
this is how it works,
with the reference of youtubePlayer view you have to setup youTubePlayerView.setOnTouchListener(this),
and implement YouTubePlayerView.OnTouchEvent on the Fragment or activity you want Click Listener ,
you'll get callback in override function onTouchReceived

### CODE REVIEW

File: core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/SixteenByNineFrameLayout.kt [hunk 1]
```diff
@@ -11,7 +12,8 @@ import androidx.annotation.RestrictTo
  * A FrameLayout with an aspect ration of 16:9, when the height is set to wrap_content.
  */
 @RestrictTo(RestrictTo.Scope.LIBRARY)
-open class SixteenByNineFrameLayout: FrameLayout {
+open class SixteenByNineFrameLayout : FrameLayout {
+
```
Comment (PierfrancescoSoffritti) [inline core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/SixteenByNineFrameLayout.kt]: Remove space.

File: core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/SixteenByNineFrameLayout.kt [hunk 2]
```diff
@@ -23,4 +25,7 @@ open class SixteenByNineFrameLayout: FrameLayout {
         } else
             super.onMeasure(widthMeasureSpec, heightMeasureSpec)
     }
+
```
Comment (PierfrancescoSoffritti) [inline core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/SixteenByNineFrameLayout.kt]: Remove spaces.

File: core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt [hunk 3]
```diff
@@ -52,16 +55,16 @@ class YouTubePlayerView(context: Context, attrs: AttributeSet? = null, defStyleA
 
         typedArray.recycle()
 
-        if(!enableAutomaticInitialization && useWebUi) {
+        if (!enableAutomaticInitialization && useWebUi) {
```
Comment (PierfrancescoSoffritti) [inline core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt]: Revert to original, this just adds noise to the pull request.

File: core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt [hunk 4]
```diff
@@ -17,18 +18,20 @@ import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.listeners.You
 import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.listeners.YouTubePlayerListener
 import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.options.IFramePlayerOptions
 import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.utils.FullScreenHelper
-import com.pierfrancescosoffritti.androidyoutubeplayer.core.ui.PlayerUiController
 import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.utils.loadOrCueVideo
+import com.pierfrancescosoffritti.androidyoutubeplayer.core.ui.PlayerUiController
 
 
-class YouTubePlayerView(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0):
+class YouTubePlayerView(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) :
```
Comment (PierfrancescoSoffritti) [inline core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt]: Revert to original, this just adds noise to the pull request.

File: core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt [hunk 5]
```diff
@@ -17,18 +18,20 @@ import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.listeners.You
...
         SixteenByNineFrameLayout(context, attrs, defStyleAttr), LifecycleObserver {
 
-    constructor(context: Context): this(context, null, 0)
-    constructor(context: Context, attrs: AttributeSet? = null): this(context, attrs, 0)
+    constructor(context: Context) : this(context, null, 0)
```
Comment (PierfrancescoSoffritti) [inline core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt]: Revert to original, this just adds noise to the pull request.

File: core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt [hunk 6]
```diff
@@ -17,18 +18,20 @@ import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.listeners.You
...
+    constructor(context: Context, attrs: AttributeSet? = null) : this(context, attrs, 0)
```
Comment (PierfrancescoSoffritti) [inline core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt]: Revert to original, this just adds noise to the pull request.

File: core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt [hunk 7]
```diff
@@ -52,16 +55,16 @@ class YouTubePlayerView(context: Context, attrs: AttributeSet? = null, defStyleA
...
 
-        if(!enableAutomaticInitialization && useWebUi) {
+        if (!enableAutomaticInitialization && useWebUi) {
             throw IllegalStateException("YouTubePlayerView: 'enableAutomaticInitialization' is false and 'useWebUi' is set to true. " +
                     "This is not possible, if you want to manually initialize YouTubePlayerView and use the web ui, " +
                     "you should manually initialize the YouTubePlayerView using 'initializeWithWebUi'")
         }
 
-        if(videoId == null && autoPlay)
+        if (videoId == null && autoPlay)
```
Comment (PierfrancescoSoffritti) [inline core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt]: Revert to original, this just adds noise to the pull request.

File: core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt [hunk 8]
```diff
@@ -52,16 +55,16 @@ class YouTubePlayerView(context: Context, attrs: AttributeSet? = null, defStyleA
...
             throw IllegalStateException("YouTubePlayerView: videoId is not set but autoPlay is set to true. This combination is not possible.")
 
-        if(!useWebUi) {
+        if (!useWebUi) {
```
Comment (PierfrancescoSoffritti) [inline core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt]: Revert to original, this just adds noise to the pull request.

File: core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt [hunk 9]
```diff
@@ -81,8 +84,8 @@ class YouTubePlayerView(context: Context, attrs: AttributeSet? = null, defStyleA
             }
         }
 
-        if(enableAutomaticInitialization) {
-            if(useWebUi) legacyTubePlayerView.initializeWithWebUi(youTubePlayerListener, handleNetworkEvents)
+        if (enableAutomaticInitialization) {
```
Comment (PierfrancescoSoffritti) [inline core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt]: Revert to original, this just adds noise to the pull request.

File: core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt [hunk 10]
```diff
@@ -81,8 +84,8 @@ class YouTubePlayerView(context: Context, attrs: AttributeSet? = null, defStyleA
...
+            if (useWebUi) legacyTubePlayerView.initializeWithWebUi(youTubePlayerListener, handleNetworkEvents)
```
Comment (PierfrancescoSoffritti) [inline core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt]: Revert to original, this just adds noise to the pull request.

File: core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt [hunk 11]
```diff
@@ -118,7 +144,7 @@ class YouTubePlayerView(context: Context, attrs: AttributeSet? = null, defStyleA
      * @see YouTubePlayerView.initialize
      */
     fun initialize(youTubePlayerListener: YouTubePlayerListener, handleNetworkEvents: Boolean) {
-        if(enableAutomaticInitialization) throw IllegalStateException("YouTubePlayerView: If you want to initialize this view manually, you need to set 'enableAutomaticInitialization' to false")
+        if (enableAutomaticInitialization) throw IllegalStateException("YouTubePlayerView: If you want to initialize this view manually, you need to set 'enableAutomaticInitialization' to false")
```
Comment (PierfrancescoSoffritti) [inline core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt]: Revert to original, this just adds noise to the pull request.

File: core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt [hunk 12]
```diff
@@ -140,7 +166,7 @@ class YouTubePlayerView(context: Context, attrs: AttributeSet? = null, defStyleA
      * @see YouTubePlayerView.initialize
      */
     fun initializeWithWebUi(youTubePlayerListener: YouTubePlayerListener, handleNetworkEvents: Boolean) {
-        if(enableAutomaticInitialization) throw IllegalStateException("YouTubePlayerView: If you want to initialize this view manually, you need to set 'enableAutomaticInitialization' to false")
+        if (enableAutomaticInitialization) throw IllegalStateException("YouTubePlayerView: If you want to initialize this view manually, you need to set 'enableAutomaticInitialization' to false")
```
Comment (PierfrancescoSoffritti) [inline core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt]: Revert to original, this just adds noise to the pull request.

File: core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt [hunk 13]
```diff
@@ -150,7 +176,7 @@ class YouTubePlayerView(context: Context, attrs: AttributeSet? = null, defStyleA
      * This function is called only once.
      */
     fun getYouTubePlayerWhenReady(youTubePlayerCallback: YouTubePlayerCallback) =
-        legacyTubePlayerView.getYouTubePlayerWhenReady(youTubePlayerCallback)
+            legacyTubePlayerView.getYouTubePlayerWhenReady(youTubePlayerCallback)
```
Comment (PierfrancescoSoffritti) [inline core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt]: Revert to original, this just adds noise to the pull request.

File: core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt [hunk 14]
```diff
@@ -200,4 +226,16 @@ class YouTubePlayerView(context: Context, attrs: AttributeSet? = null, defStyleA
 
     fun removeFullScreenListener(fullScreenListener: YouTubePlayerFullScreenListener): Boolean =
             fullScreenHelper.removeFullScreenListener(fullScreenListener)
+
+
+    /*function which sets OntouchEvent Listener*/
+    fun setOnTouchListener(onTouchEvent: OnTouchEvent) {
+        if (this.onTouchEvent == null) {
```
Comment (PierfrancescoSoffritti) [inline core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt]: Why the check for null?

File: core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt [hunk 15]
```diff
@@ -200,4 +226,16 @@ class YouTubePlayerView(context: Context, attrs: AttributeSet? = null, defStyleA
...
+            this.onTouchEvent = onTouchEvent
+        }
+    }
+
+    interface OnTouchEvent {
```
Comment (PierfrancescoSoffritti) [inline core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt]: Can't we just use a framework's listener? Like `OnTouchListener`?

File: core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt [hunk 16]
```diff
@@ -98,6 +101,29 @@ class YouTubePlayerView(context: Context, attrs: AttributeSet? = null, defStyleA
         })
     }
 
+    /*Ontouch event override function*/
+    override fun dispatchTouchEvent(event: MotionEvent): Boolean {
```
Comment (PierfrancescoSoffritti) [inline core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt]: Can't we just add an `OnTouchListener` and trigger the callback from there?

File: core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt (hunk 14 above)
Comment (mrkamiali) [reply core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt]: I case User can set multiple TouchListener that why for the sake of single reference.

File: core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt (hunk 15 above)
Comment (mrkamiali) [reply core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt]: I have tried, but I can not get any click call back in my view that why I have created my own.

File: core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt (hunk 16 above)
Comment (mrkamiali) [reply core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/views/YouTubePlayerView.kt]: Actually onTouch listener gets called in from this dispatchTouchEvent, I can send call back to OnTouchListener by checking the action Up OR Down by calling below function, 
    override fun performClick(): Boolean {
        return super.performClick()
    }
it returns the onTouchListener event in view, but in our case some how it is not returning to setOnTouchListener call back in our view that's why I have created my own callback.

**Comment by mrkamiali** [comment]
build
@PierfrancescoSoffritti  My Build is getting failed cuz of 29 SDK not found in Travis.

**Comment by PierfrancescoSoffritti** [comment]
I'll review the code over the weekend. Don't bother about Travis, I need to fix it.

**Comment by mrkamiali** [comment]
> o fix i

Sure.

**Comment by mrkamiali** [comment]
> I'll review the code over the weekend. Don't bother about Travis, I need to fix it.

sure

**Comment by PierfrancescoSoffritti** [comment]
This pull request is implementing two different new features. A pull request should be about a single new feature only.
Unfortunately I can't merge this.

**Comment by mrkamiali** [comment]
Np

**Pull request closed without merging**